lib/elixir/test/elixir/module/types/expr_test.exs

```diff
@@ -193,12 +201,95 @@ defmodule Module.Types.ExprTest do
       assert {:error, _} = quoted_expr([a], case(a, do: (:foo = b -> :bar = b)))
 
       assert quoted_expr([a], case(a, do: (:foo = b -> :foo = b))) == {:ok, :dynamic}
+
+      assert quoted_expr(
+               [a],
+               case a do
+                 :foo = b -> b
+                 :bar = b -> b
+               end
+             ) == {:ok, :dynamic}
```

Comment:
I am not sure these tests are correct. Those will be solved by variable versioning, so they won't catch an error on the verification. Instead, you want to check if the same variable outside of the case comes out as clean. Something like this:

```elixir
quoted_expr(
  [a, b],
  quote do
    case a do
      :foo when is_integer(b) -> :ok
    end
    b
  end
)
```

should still return `b` as dynamic because the clause may never be executed.

Reply:
Please ignore me, i see the correct test below. :)